File: frontend/web/features/management/types/course.ts
export type CourseValues = Record<string, string> & {
  code: string;
  name: string;
  departmentId: string;
  department: string;
  teacherId: string;
  teacher: string;
  capacity: string;
  status: string;
};

export type CourseItem = { id: string; values: CourseValues };

File: backend/src/InstituteManagement.Domain/Entities/Management/Courses/Course.cs
namespace InstituteManagement.Domain.Entities;

public sealed class Course : Entity
{
    public required string CourseCode { get; set; }
    public required string Name { get; set; }
    public int YearLevel { get; set; } = 1;
    public string Semester { get; set; } = "Semester 1";
    public Guid? DepartmentId { get; set; }
    public Department? Department { get; set; }
    public Guid? TeacherId { get; set; }
    public Teacher? Teacher { get; set; }
    public int Capacity { get; set; } = 40;
    public bool IsActive { get; set; } = true;
}
